src/drivers/distance_sensor/vl53lxx/vl53lxx.cpp

```diff
@@ -0,0 +1,1223 @@
+	_reports(nullptr),
+	_sensor_ok(false),
+	_measure_ticks(0),
+	_collect_phase(false),
+	_new_measurement(true),
+	_measurement_started(false),
+	_class_instance(-1),
+	_orb_class_instance(-1),
+	_distance_sensor_topic(nullptr),
+	_sample_perf(perf_alloc(PC_ELAPSED, "vl53lxx_read")),
+	_comms_errors(perf_alloc(PC_COUNT, "vl53lxx_com_err"))
+{
+	// up the retries since the device misses the first measure attempts
+	I2C::_retries = 2;
+
+	// enable debug() calls
+	_debug_enabled = false;
+
+	// work_cancel in the dtor will explode if we don't do this...
+	memset(&_work, 0, sizeof(_work));
+}
+
+VL53LXX::~VL53LXX()
+{
+	/* make sure we are truly inactive */
+	stop();
+
+	/* free any existing reports */
+	if (_reports != nullptr) {
+		delete _reports;
+	}
+
+	if (_distance_sensor_topic != nullptr) {
+		orb_unadvertise(_distance_sensor_topic);
+	}
+
+	if (_class_instance != -1) {
+		unregister_class_devname(RANGE_FINDER_BASE_DEVICE_PATH, _class_instance);
+	}
+
+	// free perf counters
+	perf_free(_sample_perf);
+	perf_free(_comms_errors);
+}
+
+
+int
+VL53LXX::sensorInit()
+{
+	uint8_t val = 0;
+	int ret = OK;
+	float rate_limit;
+	uint8_t rate_limit_split[2];
+
+	// I2C at 2.8V on sensor side of level shifter
+	ret |= readRegister(VHV_CONFIG_PAD_SCL_SDA_EXTSUP_HW_REG, val);
+
+	if (ret != OK) {
+		ret = PX4_ERROR;
+		return ret;
+	}
+
+	ret |= writeRegister(VHV_CONFIG_PAD_SCL_SDA_EXTSUP_HW_REG, val | 0x01);
+
+	// set I2C to standard mode
+	ret |= writeRegister(0x88, 0x00);
+
+	ret |= writeRegister(0x80, 0x01);
+	ret |= writeRegister(0xFF, 0x01);
+	ret |= writeRegister(0x00, 0x00);
+	ret |= readRegister(0x91, val);
+	ret |= writeRegister(0x00, 0x01);
+	ret |= writeRegister(0xFF, 0x00);
+	ret |= writeRegister(0x80, 0x00);
+
+	if (ret != OK) {
+		ret = PX4_ERROR;
+		return ret;
+	}
+
+	stop_variable_ = val;
+
+	// disable SIGNAL_RATE_MSRC (bit 1) and SIGNAL_RATE_PRE_RANGE (bit 4) limit checks
+	readRegister(MSRC_CONFIG_CONTROL_REG, val);
+	writeRegister(MSRC_CONFIG_CONTROL_REG, val | 0x12);
+
+	// Set signal rate limit to 0.1
+	rate_limit = 0.1 * 65536;
+	rate_limit_split[0] = (((uint16_t)rate_limit) >> 8);
+	rate_limit_split[1] = (uint16_t)rate_limit;
+
+	writeRegisterMulti(FINAL_RANGE_CONFIG_MIN_COUNT_RATE_RTN_LIMIT_REG, &rate_limit_split[0], 2);
+	writeRegister(SYSTEM_SEQUENCE_CONFIG_REG, 0xFF);
+
+	spadCalculations();
+
+	ret = OK;
+
+	return ret;
+
+}
+
+int
+VL53LXX::init()
+{
+	int ret = PX4_ERROR;
+
+	set_device_address(VL53LXX_BASEADDR);
+
+	/* do I2C init (and probe) first */
+	if (I2C::init() != OK) {
+		goto out;
+	}
+
+	/* allocate basic report buffers */
+	_reports = new ringbuffer::RingBuffer(2, sizeof(distance_sensor_s));
+
+	if (_reports == nullptr) {
+		goto out;
+	}
+
+	_class_instance = register_class_devname(RANGE_FINDER_BASE_DEVICE_PATH);
+
+	if (_class_instance == CLASS_DEVICE_PRIMARY) {
+		/* get a publish handle on the range finder topic */
+		struct distance_sensor_s ds_report;
+
+		_reports->get(&ds_report);
```

Comment:
You are reading from _reports, but nothing has been added to the ringbuffer yet.